homeassistant/components/dsmr_reader/definitions.py

```diff
@@ -27,197 +29,202 @@ def tariff_transform(value):
         "icon": "mdi:flash",
-        "unit": VOLT,
+        "device_class": VOLT,
     },
     "dsmr/reading/phase_power_current_l1": {
         "name": "Phase power current L1",
         "icon": "mdi:flash",
-        "unit": ELECTRICAL_CURRENT_AMPERE,
+        "device_class": ELECTRICAL_CURRENT_AMPERE,
     },
     "dsmr/reading/phase_power_current_l2": {
         "name": "Phase power current L2",
         "icon": "mdi:flash",
-        "unit": ELECTRICAL_CURRENT_AMPERE,
+        "device_class": ELECTRICAL_CURRENT_AMPERE,
     },
     "dsmr/reading/phase_power_current_l3": {
         "name": "Phase power current L3",
         "icon": "mdi:flash",
-        "unit": ELECTRICAL_CURRENT_AMPERE,
+        "device_class": ELECTRICAL_CURRENT_AMPERE,
+    },
+    "dsmr/reading/timestamp": {
+        "name": "Telegram timestamp",
+        "icon": "mdi:clock",
+        "device_class": DEVICE_CLASS_TIMESTAMP,
     },
     "dsmr/consumption/gas/delivered": {
         "name": "Gas usage",
         "icon": "mdi:fire",
-        "unit": VOLUME_CUBIC_METERS,
+        "device_class": VOLUME_CUBIC_METERS,
     },
     "dsmr/consumption/gas/currently_delivered": {
         "name": "Current gas usage",
         "icon": "mdi:fire",
-        "unit": VOLUME_CUBIC_METERS,
+        "device_class": VOLUME_CUBIC_METERS,
     },
     "dsmr/consumption/gas/read_at": {
         "name": "Gas meter read",
         "icon": "mdi:clock",
-        "unit": "",
+        "device_class": DEVICE_CLASS_TIMESTAMP,
     },
     "dsmr/day-consumption/electricity1": {
         "name": "Low tariff usage",
         "icon": "mdi:counter",
-        "unit": ENERGY_KILO_WATT_HOUR,
+        "device_class": ENERGY_KILO_WATT_HOUR,
     },
     "dsmr/day-consumption/electricity2": {
         "name": "High tariff usage",
         "icon": "mdi:counter",
-        "unit": ENERGY_KILO_WATT_HOUR,
+        "device_class": ENERGY_KILO_WATT_HOUR,
     },
     "dsmr/day-consumption/electricity1_returned": {
         "name": "Low tariff return",
         "icon": "mdi:counter",
-        "unit": ENERGY_KILO_WATT_HOUR,
+        "device_class": ENERGY_KILO_WATT_HOUR,
     },
     "dsmr/day-consumption/electricity2_returned": {
         "name": "High tariff return",
         "icon": "mdi:counter",
-        "unit": ENERGY_KILO_WATT_HOUR,
+        "device_class": ENERGY_KILO_WATT_HOUR,
     },
     "dsmr/day-consumption/electricity_merged": {
         "name": "Power usage total",
         "icon": "mdi:counter",
-        "unit": ENERGY_KILO_WATT_HOUR,
+        "device_class": ENERGY_KILO_WATT_HOUR,
     },
     "dsmr/day-consumption/electricity_returned_merged": {
         "name": "Power return total",
         "icon": "mdi:counter",
-        "unit": ENERGY_KILO_WATT_HOUR,
+        "device_class": ENERGY_KILO_WATT_HOUR,
     },
     "dsmr/day-consumption/electricity1_cost": {
         "name": "Low tariff cost",
         "icon": "mdi:currency-eur",
-        "unit": CURRENCY_EURO,
+        "device_class": CURRENCY_EURO,
```

Comment:
`CURRENCY_EURO` is not an official device class. Only our standard device classes are allowed:
https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Reply:
Yes you are right! It should be fixed now 👍